src/main/java/at/hannibal2/skyhanni/features/dungeon/CroesusChestTracker.kt

```diff
@@ -223,10 +243,76 @@ object CroesusChestTracker {
         event.stackTip = "§a✔"
     }
 
+    @HandleEvent
+    fun onKuudraComplete(event: KuudraCompleteEvent) {
+        addCroesusChest("T${event.kuudraTier}")
+    }
+
     @HandleEvent
     fun onDungeonComplete(event: DungeonCompleteEvent) {
         if (event.floor == "E") return
-        croesusChests?.add(0, DungeonRunInfo(event.floor))
+        addCroesusChest(event.floor)
+    }
+
+    @Suppress("MaxLineLength")
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onGuiRender(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (config.croesusOverlayKuudra && KuudraApi.inKuudra) renderChestOverlay()
+        if (config.croesusOverlay && (SkyBlockUtils.graphArea == "Forgotten Skull" || SkyBlockUtils.scoreboardArea == "Dungeon Hub")) renderChestOverlay()
```

Comment:
the scoreboardArea check here is not good enough. there are no sub areas in dh, this currently shows everywhere. either replace it with `IslandType.DUNGEON_HUB.isCurrent()` or use a different way to only show the display around mort/croesus.